docs/providers/GeoNames.rst

```diff
@@ -51,6 +52,37 @@ They are all supported
     >>> g.population
     19274244
 
+Proximity
+---------
+
+Note the extra parameters 'east', 'west', 'north', 'south' : they provide the functionnality 'proximity that can be found on some other providers (e.g Mapbox)
+
+
+.. code-block:: python
+
+    >>> g = geocoder.geonames('Paris', key='<USERNAME>', east=-96.0, west=-95.0, north=33.5, south=33.0)
```

Comment:
I would recommend we also add `bbox` as another parameter to define providers which support bounds.

Since BBox is a standard in the GeoJSON spec, it's widely adopted as `[west, south, east, north]`.

I don't think much providers have an additional `bbox` parameter, but if we do we should include in future releases.